### ISSUE TITLE
Update unit security roles

### ISSUE THREAD

**Issue by brent-hartwig** [project-lux/lux-marklogic#280]
> [!NOTE]
> Nothing for UAT.  Developer needs to verify.

> [!NOTE]
> Revised description and scope based on need and requirements known on 3 Sep 24.

We already have reader and endpoint consumer roles for YCBA, YPM, and YUAG (defined [here](https://github.com/project-lux/lux-marklogic/tree/main/src/main/ml-config/base/security/roles)).

As of 3 Sep 24, the `admin.sources` array could also have values for ILS, IPCH, and PMC (lowercased), thereby necessitating at least reader roles for them.  Note: "ILS" was selected over "YUL".

Only YPM requires an endpoint consumer role, which also means only YPM needs unit-specific configurations (#277).

ML Gradle configuration file changes:

- [x] Add tenant-specific reader roles for a couple values the data pipeline may include and the transform doesn't otherwise like: 'create' and 'update'
- [x] Add tenant-specific reader roles for ILS, IPCH, and PMC.
- [x] Add tenant-specific endpoint consumer role for IPCH.
- [x] Remove the tenant-specific endpoint consumer roles for YCBA and YUAG.

(Of the units, only IPCH and YPM are to have endpoint consumer roles at this time.)

Please also see [this comment](https://github.com/project-lux/lux-marklogic/issues/280#issuecomment-2329345184) for manual changes that may be required in each environment and tenant.

cc: @azaroth42, @roamye, and @clarkepeterf

**Comment by clarkepeterf** [comment]
@brent-hartwig not sure if we want to lump this together, but I've also been meaning to make this change to only use allowed values in `admin.sources`, as set via a gradle property

https://github.com/project-lux/lux-marklogic/compare/a8c8016..ignore-create-update-document-transforms?expand=1

with some additional changes as you suggested:

> 1. Changing the property name to allowedSourceNames or allowedSources, to be closer to the property name.
> 2. Lowercasing and removing all spaces from the property value before splitting.  The latter makes the property value more readable: `allowedSources=ypm, ycba, etc..` This would mean the code wouldn't support names with spaces in them.
> 3. Lowercasing `name` before the `includes` check.

**Comment by brent-hartwig** [comment]
Looks good, @clarkepeterf.  For #277, I am introducing the `endpointAccessUnitNames` property and believe it serves a unique purpose.  Below is the inline documentation.  There's more in the [Unit Portal](https://github.com/project-lux/lux-marklogic/blob/277-unit-specific-config-files/docs/lux-backend-security-and-software.md#unit-portals) documentation (link is to #277's branch).  I changed my mind a bit since providing feedback to you: I'm purposely not lowercasing the values and am supporting anything between `lux-` and `-endpoint-consumer` in the role name; please see [getCurrentUserUnitName](https://github.com/project-lux/lux-marklogic/blob/277-unit-specific-config-files/src/main/ml-modules/root/lib/unitLib.mjs#L30-L58) (another #277 branch link).  I plan to PR this today so perhaps we can lock in identical treatment of these two properties beforehand.

```
# List the units that intend to consume the backend endpoints *and* expect their data and configuration
# to be restricted to their data and shared data.  This will likely be a subset of values found in the 
# /admin/sources array.
#
# Unit names are to be separated by commas and match "[unitName]" in its "lux-[unitName]-endpoint-consumer"
# role name (case-sensitive).  Spaces may be included for readability; e.g.:
#
# endpointAccessUnitNames=pmc, ypm
#
endpointAccessUnitNames=ypm
```

**Comment by brent-hartwig** [comment]
@roamye, @jffcamp, and @prowns, I needed to create these roles for tonight's load into SBX, and thus bypassed prioritization review.

**Comment by jffcamp** [comment]
[like]  Campbell, Jeffrey reacted to your message:
________________________________

**Comment by brent-hartwig** [comment]
Implemented in PRs #307, #308, and #310.

**Comment by brent-hartwig** [comment]
Manual changes made in Blue today, after it rejoined TST.  Closing this ticket.
